db.go

```diff
@@ -676,7 +676,43 @@ const (
 // Sync syncs database content to disk. This function provides
 // more control to user to sync data whenever required.
 func (db *DB) Sync() error {
-	return db.vlog.sync()
+	/**
+	Make an attempt to sync both the logs, the active memtable's WAL and the vLog (1847).
+	Cases:
+	- All_ok			:: If both the logs sync successfully.
+
+	- Entry_Lost		:: If an entry with a value pointer was present in the active memtable's WAL,
+						:: and the WAL was synced but there was an error in syncing the vLog.
+						:: The entry will be considered lost and this case will need to be handled during recovery.
+
+	- Entries_Lost		:: If there were errors in syncing both the logs, multiple entries would be lost.
+
+	- Entries_Lost      :: If the active memtable's WAL is not synced but the vLog is synced, it will
+						:: result in entries being lost because recovery of the active memtable is done from its WAL.
+						:: Check `UpdateSkipList` in memtable.go.
+
+	- Nothing_lost		:: If an entry with its value was present in the active memtable's WAL, and the WAL was synced,
+						:: but there was an error in syncing the vLog.
+						:: Nothing is lost for this very specific entry because the entry is completely present in the memtable's WAL.
+
+	- Partially_lost    :: If entries were written partially in either of the logs, the logs will be truncated during recovery.
+						:: As a result of truncation, some entries might be lost. Assume that 4KB of data is to be synced and
+					    :: invoking `Sync` results only in syncing 3KB of data and then the machine shuts down or the disk failure happens,
+	                    :: this will result in partial writes. [[This case needs verification]]
+	*/
+
+	var err error
+
+	memtableSyncError := db.mt.SyncWAL()
```

Comment:
we probably should acquire a lock here

Reply:
Sure, but why was the original sync of vlog without locks?

Reply:
For now, I have changed the code to use lock (`the write lock`).

Reply:
I think for vlog, we acquire the lock inside the sync function. We need to do something for memtable.

Reply:
Ok, wait. Let me change the code to get `RLock` and unlock before I sync vlog